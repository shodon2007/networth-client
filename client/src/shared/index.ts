export * from './context/themeProvider'
export * from './ui'
 


