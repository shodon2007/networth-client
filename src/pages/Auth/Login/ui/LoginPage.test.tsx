import userEvent, {UserEvent} from "@testing-library/user-event";
import {renderTestApp} from "src/shared/tests";

describe("teting LoginPage.tsx", () => {
	let user: UserEvent;
	beforeAll(() => {
		user = userEvent.setup();
	});
	test("snapshot testing", async () => {
		const screen = renderTestApp("/login");
		expect(await screen.findByTestId("login-form")).toBeInTheDocument();
		expect(screen.getByTestId("login-form")).toMatchSnapshot();
	});
	test("Testing registration link", async () => {
		const screen = renderTestApp("/login");
		const registrationLink = await screen.findByTestId("registration-link");
		expect(await screen.findByTestId("login-form")).toBeInTheDocument();
		expect(screen.queryByTestId("registration-form")).not.toBeInTheDocument();
		await user.click(registrationLink);
		expect(await screen.findByTestId("registration-form")).toBeInTheDocument();
		expect(screen.queryByTestId("login-form")).not.toBeInTheDocument();
	});

	test("Testing login", async () => {
		const screen = renderTestApp("/login");
		const loginBtn = await screen.findByTestId("login-button");
		const loginEmailInput = await screen.findByTestId("email");
		const loginPasswordInput = screen.getByTestId("password");
		screen.debug();
		await user.type(loginEmailInput, "about@shodon.ru");
		await user.type(loginPasswordInput, "shodon2007");

		await user.click(loginBtn);
		screen.debug();
	});
});
