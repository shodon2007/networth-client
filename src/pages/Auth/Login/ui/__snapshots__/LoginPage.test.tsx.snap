// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teting LoginPage.tsx snapshot testing 1`] = `
<form
  data-testid="login-form"
>
  <div
    class="block default page"
    data-testid="block"
  >
    <h1
      class="title MIDDLE"
    >
      Войти
    </h1>
    <div
      class="inputs"
    >
      <div
        data-testid="input-parent"
      >
        <input
          class="input medium"
          data-testid="email"
          name="email"
          placeholder="Введите почту"
          value=""
        />
      </div>
      <div
        data-testid="input-parent"
      >
        <input
          class="input medium"
          data-testid="password"
          name="password"
          placeholder="Введите пароль"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="bottom"
    >
      <button
        class="button submit"
        data-testid="login-button"
        type="submit"
      >
        Войти
      </button>
      <a
        class="link"
        data-testid="registration-link"
        href="/registration"
      >
        Создать аккаунт
      </a>
    </div>
  </div>
</form>
`;
